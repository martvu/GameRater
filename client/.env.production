VITE_SERVER_URI=http://it2810-48.idi.ntnu.no:4000/graphql