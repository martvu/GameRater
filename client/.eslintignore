dist
.eslintrc.js
.prettierrc.js
src/gql/