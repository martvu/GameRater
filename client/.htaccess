<IfModule mod_rewrite.c>
  RewriteEngine On
  RewriteBase /project2/
  RewriteRule ^index\.html$ - [L]
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteRule . /project2/index.html [L]
</IfModule>
