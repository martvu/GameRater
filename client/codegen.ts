import type { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
  schema: '../server/src/schema.graphql',
  documents: ['src/**/*.{ts,tsx}'],
  generates: {
    './src/gql/': {
      preset: 'client',
      plugins: [],
      overwrite: true,
      presetConfig: {
        gqlTagName: 'gql',
        fragmentMasking: false,
      },
    },
  },
};
export default config;
