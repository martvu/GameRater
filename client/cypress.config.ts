import { defineConfig } from 'cypress';
import { connect, disconnect } from './cypress/support/db';

export default defineConfig({
  e2e: {
    baseUrl: 'http://localhost:5173/project2',
    setupNodeEvents(on) {
      // This task cleans the reviews and users collections in the database
      on('task', {
        async clearDB() {
          const db = await connect();
          const reviews = db.collection('reviews');
          await reviews.deleteMany({});
          await reviews.dropIndexes();
          const users = db.collection('users');
          await users.deleteMany({});
          await users.dropIndexes();
          await disconnect();
          return null;
        },
      });
    },
  },
});
