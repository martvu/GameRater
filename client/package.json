{
  "name": "prosjekt-2",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "npm run generate && vite --host",
    "build": "tsc && vite build",
    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "preview": "vite preview",
    "test": "vitest",
    "coverage": "vitest run --coverage",
    "generate": "graphql-codegen",
    "cypress:open": "cypress open"
  },
  "dependencies": {
    "@apollo/client": "^3.8.6",
    "@hookform/resolvers": "^3.3.2",
    "@radix-ui/react-avatar": "^1.0.4",
    "@radix-ui/react-checkbox": "^1.0.4",
    "@radix-ui/react-dialog": "^1.0.5",
    "@radix-ui/react-dropdown-menu": "^2.0.6",
    "@radix-ui/react-label": "^2.0.2",
    "@radix-ui/react-scroll-area": "^1.0.5",
    "@radix-ui/react-select": "^2.0.0",
    "@radix-ui/react-slot": "^1.0.2",
    "@radix-ui/react-toast": "^1.1.5",
    "@radix-ui/react-tooltip": "^1.0.7",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.0.0",
    "graphql": "^16.8.1",
    "lucide-react": "^0.284.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-hook-form": "^7.47.0",
    "react-infinite-scroll-component": "^6.1.0",
    "react-router-dom": "^6.16.0",
    "recoil": "^0.7.7",
    "tailwind-merge": "^1.14.0",
    "tailwindcss-animate": "^1.0.7",
    "zod": "^3.22.4"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "^5.0.0",
    "@graphql-codegen/client-preset": "^4.1.0",
    "@testing-library/jest-dom": "^6.1.3",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.5.1",
    "@types/jest": "^29.5.5",
    "@types/node": "^20.8.6",
    "@types/react": "^18.2.15",
    "@types/react-dom": "^18.2.7",
    "@typescript-eslint/eslint-plugin": "^6.0.0",
    "@typescript-eslint/parser": "^6.0.0",
    "@vitejs/plugin-react-swc": "^3.3.2",
    "@vitest/coverage-v8": "^0.34.6",
    "autoprefixer": "^10.4.16",
    "cypress": "^13.5.1",
    "eslint": "^8.45.0",
    "eslint-config-prettier": "^9.0.0",
    "eslint-plugin-prettier": "^5.0.1",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.3",
    "jsdom": "^22.1.0",
    "postcss": "^8.4.31",
    "prettier": "^3.0.3",
    "prettier-plugin-tailwindcss": "^0.5.5",
    "tailwindcss": "^3.3.3",
    "typescript": "^5.2.2",
    "vite": "^4.4.5",
    "vitest": "^0.34.6"
  }
}
