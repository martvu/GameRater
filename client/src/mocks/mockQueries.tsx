import {
  ADD_FAVORITES,
  CREATE_REVIEW,
  REMOVE_FAVORITES,
  SIGN_IN_OR_CREATE_USER,
} from '@/lib/mutations';
import {
  GET_GAMES,
  GET_GAME,
  GET_FILTERS,
  GET_GAME_PLATFORMS,
  GET_SEARCH_SUGGESTIONS,
} from '@/lib/queries';

const getGamesMock = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: undefined,
      userId: '',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            user_rating: 4.5,
          },
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getGamesMockedKeyword = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: 'mockedKeyword',
      userId: '',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            user_rating: 4.5,
          },
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getGamesSignedInMock = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: 'mockedKeyword',
      userId: '1',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            platforms: [1, 2],
            genres: [3, 4],
          },
          // Add more game objects as needed
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getFiltersMock = {
  request: {
    query: GET_FILTERS,
    variables: {},
  },
  result: {
    data: {
      getGenres: [
        { id: 1, name: 'Genre 1' },
        { id: 2, name: 'Genre 2' },
      ],
      getPlatforms: [
        { id: 1, name: 'Platform 1' },
        { id: 3, name: 'Platform 3' },
        { id: 4, name: 'PC (Microsoft Windows)' },
        { id: 2, name: 'Nintendo Switch' },
        { id: 5, name: 'PlayStation 5' },
        { id: 6, name: 'Xbox Series X|S' },
      ],
    },
  },
};

const getGamePlatformsMock = {
  request: {
    query: GET_GAME_PLATFORMS,
    variables: { id: '1' },
  },
  result: {
    data: {
      getGame: {
        name: 'Example Game',
        platforms: [
          {
            id: 3,
            name: 'Platform 3',
          },
          {
            id: 4,
            name: 'PC (Microsoft Windows)',
          },
        ],
      },
    },
  },
};

const signInMock = {
  request: {
    query: SIGN_IN_OR_CREATE_USER,
    variables: {
      userInput: {
        username: 'testuser',
      },
    },
  },
  result: {
    data: {
      signInOrCreateUser: {
        _id: '1',
        username: 'testuser',
        favorites: [],
        reviews: [],
      },
    },
  },
};

const getGameMock = {
  request: {
    query: GET_GAME,
    variables: {
      id: '1',
      limit: 5,
      offset: 0,
      username: '',
    },
  },
  result: {
    data: {
      getGame: {
        _id: '1',
        name: 'Example Game',
        summary: 'This is an example game summary.',
        imageId: 'image1',
        first_release_date: 1620000000,
        aggregatedRating: 85,
        user_rating: 4.5,
        platforms: [{ name: 'PC' }],
        genres: [{ name: 'Adventure' }],
        reviews: {
          count: 2,
          reviews: [
            {
              _id: 'review1',
              user: 'user1',
              title: 'Great Game',
              content: 'This is a great game because...',
              rating: 5,
              platform: 'PC',
              gameID: '1',
            },
            {
              _id: 'review2',
              user: 'user2',
              title: 'Good Game',
              content: 'This is a good game because...',
              rating: 4,
              platform: 'PC',
              gameID: '1',
            },
          ],
          userHasReviewed: false,
        },
      },
    },
  },
};

const getGameMock2 = {
  request: {
    query: GET_GAME,
    variables: {
      id: '1',
      limit: 5,
      offset: 0,
      username: 'testUser',
    },
  },
  result: {
    data: {
      getGame: {
        _id: '1',
        name: 'Example Game',
        summary: 'This is an example game summary.',
        imageId: 'image1',
        first_release_date: 1620000000,
        aggregatedRating: 85,
        user_rating: 4.5,
        platforms: [{ name: 'PC' }],
        genres: [{ name: 'Adventure' }],
        reviews: {
          count: 2,
          reviews: [
            {
              _id: 'review1',
              user: 'user1',
              title: 'Great Game',
              content: 'This is a great game because...',
              rating: 5,
              platform: 'PC',
              gameID: '1',
            },
            {
              _id: 'review2',
              user: 'user2',
              title: 'Good Game',
              content: 'This is a good game because...',
              rating: 4,
              platform: 'PC',
              gameID: '1',
            },
          ],
          userHasReviewed: false,
        },
      },
    },
  },
};
const addFavoritesMock = {
  request: {
    query: ADD_FAVORITES,
    variables: { username: 'testUser', gameID: '1' },
  },
  result: {
    data: {
      addFavorites: {
        _id: '1',
        username: 'testUser',
        favorites: [{ _id: '1' }],
      },
    },
  },
};

const removeFavoritesMock = {
  request: {
    query: REMOVE_FAVORITES,
    variables: { username: 'testUser', gameID: '1' },
  },
  result: {
    data: {
      removeFavorites: {
        _id: '1',
        username: 'testUser',
        favorites: [],
      },
    },
  },
};

const createReviewMock = {
  request: {
    query: CREATE_REVIEW,
    variables: {
      reviewInput: {
        title: 'Great Game',
        content: 'This is a great game because...',
        rating: 5,
        platform: 'PC (Microsoft Windows)',
        user: 'testUser',
        gameID: '1',
      },
    },
  },
  result: {
    data: {
      createReview: {
        _id: 'review1',
        user: 'testUser',
        title: 'Great Game',
        content: 'This is a great game because...',
        rating: 5,
        platform: 'PC (Microsoft Windows)',
        gameID: '1',
      },
    },
  },
};

const searchbarSuggestionsMock = {
  request: {
    query: GET_SEARCH_SUGGESTIONS,
    variables: {
      query: 'example',
    },
  },
  result: {
    data: {
      getSearchSuggestions: [
        {
          _id: '1',
          name: 'Example Game',
          cover_image_id: 'cover1',
        },
        {
          _id: '2',
          name: 'Example Game 2',
          cover_image_id: 'cover2',
        },
      ],
    },
  },
};

const allMocks = [
  createReviewMock,
  getGamePlatformsMock,
  getGamesMockedKeyword,
  addFavoritesMock,
  removeFavoritesMock,
  getGamesMock,
  getFiltersMock,
  signInMock,
  getGamesSignedInMock,
  getGameMock,
  getGameMock2,
  searchbarSuggestionsMock,
  // Add more mocks as needed for other queries
];

export {
  allMocks,
  getGamesMock,
  getGamesSignedInMock,
  getFiltersMock,
  signInMock,
};
