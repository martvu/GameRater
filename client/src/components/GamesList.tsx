import { useQuery } from '@apollo/client';
import { gql } from '../gql/';
import { GameCard } from './GameCard';
import { Game } from '@/gql/graphql';
import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
import {
  genresListState,
  platformsListState,
  selectedGenresState,
  selectedPlatformsState,
  showFavoritesState,
  showReviewedGamesState,
  sortByState,
  userState,
} from '@/state/atoms';
import { pageState } from '@/state/atoms';
import Pagination from './Pagination';
import { useEffect } from 'react';
import Loading from './Loading';
import { useParams } from 'react-router-dom';

const GET_GAMES = gql(`
  query Search($userId: String, $showFavorites: Boolean, $showReviewedGames: Boolean, $query: String, $limit: Int!, $offset: Int!, $sortBy: GameSortInput, $platforms: [Int!], $genres: [Int!]) {
    search(userId: $userId, showFavorites: $showFavorites, showReviewedGames: $showReviewedGames, query: $query, limit: $limit, offset: $offset, sortBy: $sortBy, platforms: $platforms, genres: $genres) {
      count
      games{
        _id
        aggregated_rating
        first_release_date
        summary
        cover_image_id
        name
      }
      filters {
        genres 
        platforms
      }
    }
  }
`);

export default function GamesList() {
  const { keyword } = useParams<{ keyword?: string }>();
  const [currentPage, setCurrentPage] = useRecoilState(pageState);
  const [sortBy, setSortBy] = useRecoilState(sortByState);
  const selectedPlatforms = useRecoilValue(selectedPlatformsState);
  const selectedGenres = useRecoilValue(selectedGenresState);
  const user = useRecoilValue(userState);
  const showFavorites = useRecoilValue(showFavoritesState);
  const showReviewedGames = useRecoilValue(showReviewedGamesState);
  const setGenresList = useSetRecoilState(genresListState);
  const setPlatformsList = useSetRecoilState(platformsListState);
  const limit = 24;

  // Ensure that keyword is a string, even if it's an empty string.
  const { loading, error, data, refetch } = useQuery(GET_GAMES, {
    variables: {
      limit,
      offset: limit * (currentPage - 1),
      sortBy: sortBy,
      platforms: selectedPlatforms,
      genres: selectedGenres,
      query: keyword,
      userId: user?._id,
      showFavorites,
      showReviewedGames,
    },
  });

  useEffect(() => {
    refetch();
  }, [showFavorites, showReviewedGames, refetch]);

  useEffect(() => {
    const sortByFromStorage = localStorage.getItem('selectedSortBy');
    if (sortByFromStorage) {
      setSortBy(JSON.parse(sortByFromStorage));
    }
  }, []);

  useEffect(() => {
    if (data?.search.filters) {
      setGenresList(data.search.filters.genres as number[]);
      setPlatformsList(data.search.filters.platforms as number[]);
    }
  }, [data?.search.filters]);

  useEffect(() => {
    // Scroll to top of the page
    window.scrollTo(0, 0);
  }, [currentPage]);

  if (loading) return <Loading />;
  if (error) return <div>Error: {error.message}</div>;

  return (
    <main className="mt-1">
      <div className="flex items-center gap-2 pb-4 text-muted-foreground opacity-80">
        <p className="w-[100px] text-left text-sm">
          {data?.search.count} results
        </p>
      </div>
      <ul className="grid grid-cols-[repeat(auto-fill,minmax(175px,1fr))] sm:grid-cols-[repeat(auto-fill,minmax(260px,1fr))] justify-center gap-1 sm:gap-4">
        {data?.search.games?.map((game: Game | null | undefined) => (
          <li className="m-1 flex justify-center" key={game?._id}>
            {game && <GameCard game={game} />}
          </li>
        ))}
      </ul>
      {data?.search.games?.length === 0 && (
        <div className="flex w-full flex-col items-center justify-center">
          <p className="text-foreground">No games found</p>
        </div>
      )}
      <div className="my-2 flex w-full justify-center">
        <Pagination
          currentPage={currentPage}
          setCurrentPage={setCurrentPage}
          pages={Math.ceil((data?.search.count || 1) / limit) || 1}
        />
      </div>
    </main>
  );
}
