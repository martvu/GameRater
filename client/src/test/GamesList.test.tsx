import { render, screen, waitFor } from '@testing-library/react';
import { describe, it, vi } from 'vitest';
import { allMocks } from '../mocks/mockQueries';
import GamesList from '@/components/GamesList';
import { MockedProvider } from '@apollo/client/testing';
import { RecoilRoot } from 'recoil';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '@/components/theme/ThemeProvider';

beforeAll(() => {
  // Mock `window.scrollTo` to accept any arguments
  window.scrollTo = vi.fn(() => {});
});

// Mock 'react-router-dom'
vi.mock('react-router-dom', async () => {
  const actualModule =
    await vi.importActual<typeof import('react-router-dom')>(
      'react-router-dom'
    );

  return {
    ...actualModule,
    useParams: vi.fn(() => ({ keyword: 'mockedKeyword' })),
  };
});

const renderGamesList = () => {
  return render(
    <MockedProvider mocks={allMocks} addTypename={false}>
      <RecoilRoot>
        <ThemeProvider defaultTheme="dark" storageKey="vite-ui-theme">
          <MemoryRouter initialEntries={['/project2']}>
            <GamesList />
          </MemoryRouter>
        </ThemeProvider>
      </RecoilRoot>
    </MockedProvider>
  );
};

describe('GameList Component', () => {
  it('renders the GamesList component with mocked GraphQL data', async () => {
    renderGamesList();
    // Wait for the mocked data to be displayed  
    await waitFor(() => {
      expect(screen.getByText('Example Game')).toBeInTheDocument();
      expect(screen.getByText('1 results')).toBeInTheDocument();
      expect(screen.getByText('4.5')).toBeInTheDocument();
      expect(screen.getByText('85')).toBeInTheDocument();
    });

  });
});
