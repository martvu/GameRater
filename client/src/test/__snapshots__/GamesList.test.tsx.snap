// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameList Component > matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="mt-1"
  >
    <article
      class="flex items-center gap-2 pb-4 text-muted-foreground opacity-80"
    >
      <p
        class="w-[100px] text-left text-sm"
      >
        1 results
      </p>
    </article>
    <ul
      class="grid grid-cols-[repeat(auto-fill,minmax(175px,1fr))] justify-center gap-1 sm:grid-cols-[repeat(auto-fill,minmax(260px,1fr))] sm:gap-4"
    >
      <li
        class="m-1 flex justify-center"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-lg h-[310px] min-w-[175px] max-w-[262px] overflow-visible p-0 duration-300 hover:scale-105 sm:h-[410px] sm:min-w-[262px]"
        >
          <a
            aria-label="Link to Example Game detail page"
            data-testid="game-card-link"
            href="/game/1"
          >
            <div
              class="flex-col flex h-[235px] w-[176px] flex-1 items-center justify-center overflow-hidden rounded-t-lg sm:mb-2 sm:h-[320px] sm:w-[260px]"
            >
              <div
                class="relative h-full w-full object-cover"
              >
                <div
                  class="absolute inset-0 z-10 flex items-center justify-center"
                >
                  <img
                    alt="Fallback cover image"
                    class="w-1/2 object-cover opacity-100 transition-opacity duration-500 ease-in-out"
                    src="/project2/src/assets/img-fallback.svg"
                  />
                </div>
                <img
                  alt="Game cover image"
                  class="absolute inset-0 h-full w-full object-cover transition-opacity duration-500 ease-in-out z-0 opacity-0"
                  loading="lazy"
                  src="https://images.igdb.com/igdb/image/upload/t_cover_big/cover1.jpg"
                />
              </div>
            </div>
            <div
              class="pt-0 relative px-3 text-start"
            >
              <h3
                class="font-semibold tracking-tight line-clamp-1 max-w-[150px] py-1 text-sm duration-300 hover:opacity-50 sm:my-1 sm:max-w-[240px] sm:text-lg"
                data-state="closed"
              >
                Example Game
              </h3>
            </div>
          </a>
          <div
            class="pt-0 relative mt-auto flex h-[40px] w-full items-center pl-3 pr-2 sm:pb-2"
          >
            <div
              class="absolute left-2 flex items-center"
            >
              <svg
                aria-label="4.5 out of 5 stars"
                class="mr-1 h-5 fill-yellow-400 text-yellow-400"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <p>
                4.5
              </p>
              <p
                aria-label="undefined reviews"
                class="ml-1 font-light text-muted-foreground opacity-80"
              >
                 
              </p>
            </div>
            <div
              class="absolute right-1 flex items-center"
            >
              <div
                aria-label="metascore"
                class="flex items-center justify-center rounded-md border border-white text-sm text-white bg-green-700 h-7 w-7 px-1"
              >
                85
              </div>
              <button
                aria-label="Favorite game button"
                class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-full z-10"
                data-testid="favorite-btn"
              >
                <svg
                  class="none"
                  data-testid="heart-icon"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <div
      class="my-2 flex w-full justify-center pt-2"
    />
  </section>
</DocumentFragment>
`;
